Snippets/TransactionalSession/TransactionalSession_1/MongoDBConfig.cs

```diff
@@ -0,0 +1,49 @@
+﻿using NServiceBus;
+using NServiceBus.ObjectBuilder;
+using NServiceBus.TransactionalSession;
+using System.Threading.Tasks;
+
+namespace TransactionalSession_1
+{
+    public class MongoDBConfig
+    {
+        public void Configure(EndpointConfiguration config)
+        {
+            #region enabling-transactional-session-mongo
+
+            config.UsePersistence<MongoPersistence>().EnableTransactionalSession();
+
+            #endregion
+        }
+
+        public async Task OpenDefault(IBuilder builder)
+        {
+            #region open-transactional-session-mongo
+
+            using var childScope = builder.CreateChildBuilder();
+            var session = childScope.Build<ITransactionalSession>();
+            await session.Open(new MongoOpenSessionOptions());
```

Comment:
ConfigureAwait?

Reply:
Isn't this one of the persisters like NHibernate where we would recommend using the `Open` extension?